src/parsers/manifest/metaplaylist/metaplaylist_parser.ts

```diff
@@ -293,13 +293,17 @@ function createManifest(
   }
 
   const time = performance.now();
+  const isLastPeriodKnown = !isDynamic ||
+                            mplData.pollInterval === undefined &&
+                            manifests[manifests.length - 1]?.isLastPeriodKnown;
```

Comment:
`isLastPeriodKnown` is potentially `undefined` if there's no element in `manifests`, no?